bosco-override-setup.sh

```diff
@@ -20,6 +20,12 @@ function errexit {
 GIT_ENDPOINT=$1
 RESOURCE_NAME=$2
 
+# pre-scan the Git repo's host key
+if [[ $GIT_ENDPOINT =~ ^[A-Za-z0-9]+@ ]]; then
```

Comment:
```suggestion
if [[ $GIT_ENDPOINT =~ ^[-A-Za-z0-9]+@ ]]; then
```

I believe hyphens are allowed in github usernames.  (Technically, not leading or trailing or multiple consecutive hyphens, but maybe it's not necessary to make the regex absolutely precise about that.)

Reply:
Relatedly is this just for github repos or more generally for any git repo endpoint?

Reply:
It's intended for any